### PR TITLE
feat: Implemented Djed Tefnut smart contracts with ETH backing

### PR DESCRIPTION
**Pull request by negimox**
Implemented Djed Tefnut smart contracts with ETH backing. For issue #37 

## Summary by CodeRabbit

* **New Features**
  * Introduces an on-chain stability system: buy/sell stable and reserve coins with oracle-backed dynamic pricing, treasury fee routing, and transaction limits.
* **Developer / Deployment**
  * Adds centralized deployment parameters and a deployment script to simplify network-specific launches and initial funding.
* **Testing / Mocks**
  * Adds a controllable mock oracle and a comprehensive test suite covering trading flows, fees, limits, pricing, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/DjedTefnut.sol [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    function rcBuyingPrice(uint256 _scPrice, uint256 currentPaymentAmount) internal view returns (uint256) {
+        return reserveCoin.totalSupply() == 0
+            ? RC_INITIAL_PRICE
+            : Math.max(rcTargetPrice(_scPrice, currentPaymentAmount), RC_MIN_PRICE);
+    }
+
+    function transferEth(address receiver, uint256 amount) internal {
+        (bool success,) = payable(receiver).call{value: amount}("");
+        require(success, "Transfer failed.");
+    }
```
[the remark this answers is not in the document]
Comment (negimox) [reply src/DjedTefnut.sol:217]: Keeping Push payments to maintain consistency with the original Djed protocol architecture.

File: src/DjedTefnut.sol [hunk 2]
```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: AEL
+pragma solidity ^0.8.0;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {Coin} from "./Coin.sol";
+import {IOracleShu} from "./IOracleShu.sol";
+
```
[the remark this answers is not in the document]
Comment (negimox) [reply src/DjedTefnut.sol:8]: Isnt this incorrect?

**Comment by negimox** [comment]
Hey, @yogesh0509 could you please if the above code rabbit actionable items? I believe they are incorrect could you provide your inputs? As iam utilizing the correct AEL license

**Comment by yogesh0509** [comment]
Hi @negimox , thank you for taking your time to work on this issue. Unfortunately, we have decided to go with another PR for this issue. Please continue working with AOSSIE in future. Thank you!
